File: backend/app/api/log.py
from fastapi import APIRouter

router = APIRouter()

@router.get("/logs")
async def get_logs():
    """
    查询系统日志接口。
    """
    # TODO: 查询日志
    return {"logs": ["日志1（示例）", "日志2（示例）"]}

@router.post("/alert")
async def send_alert(message: str):
    """
    发送告警接口。
    """
    # TODO: 实现告警推送
    return {"msg": f"告警已发送：{message}"} 

File: frontend/src/api/log.js
import request from '@/utils/request'

export function getLogs(query) {
  return request({
    url: '/log/',
    method: 'get',
    params: { search: query }
  })
}

export function deleteLog(logId) {
  return request.delete(`/log/${logId}`)
}

export function createLog({ logtype, description }, screenshot, video) {
  const formData = new FormData()
  formData.append('logtype', logtype)
  if (description) formData.append('description', description)
  if (screenshot) formData.append('screenshot', screenshot, 'screenshot.png')
  if (video) formData.append('video', video, 'video.webm')
  return request.post('/log/', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export function updateLog(logId, description) {
  return request.put(`/log/${logId}`, { description })
}

export function sendAlert(message) {
  return request.post('/log/alert', { message })
} 